src/HomeWork/hw4/MyLinkedList.java

```diff
@@ -0,0 +1,163 @@
+package HomeWork.hw4;
+
+public class MyLinkedList<T> implements MyList<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
+    @Override
+    public void add(T value) {
+        Node<T> addNode;
+        if (size == 0) {
+            addNode = new Node<>(null, null, value);
+            firstNode = addNode;
+            lastNode = addNode;
+            size++;
+        } else {
+            addNode = new Node<>(null, lastNode, value);
+            lastNode.next = addNode;
+            lastNode = addNode;
+            size++;
+        }
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        } else {
+            int currentIndex = 0;
+            Node<T> addNode;
+            Node<T> currentNode = firstNode;
+            while (currentIndex < index) {
+                currentNode = currentNode.next;
+                currentIndex++;
+            }
+            if (index == 0) {
+                addNode = new Node<>(firstNode, null, value);
+                firstNode.prev = addNode;
+                firstNode = addNode;
+            } else if (index == size) {
+                addNode = new Node<>(null, lastNode, value);
+                lastNode.next = addNode;
+                lastNode = addNode;
+            } else {
+                addNode = new Node<>(currentNode, currentNode.prev, value);
+                currentNode.prev.next = addNode;
+                currentNode.prev = addNode;
+            }
+            size++;
+        }
+    }
+
+    @Override
+    public void addAll(MyList<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        return getNode(index).value;
+    }
+
+    private Node<T> getNode(int index) {
+        int currentIndex = 0;
+        Node<T> getNode = firstNode;
+        while (currentIndex < index) {
+            getNode = getNode.next;
+            currentIndex++;
+        }
+        return getNode;
+    }
+
+    @Override
+    public void set(T value, int index) {
+        getNode(index).value = value;
+    }
+
+    @Override
+    public T remove(int index) {
+        if (size == 0) {
+            throw new NegativeArraySizeException();
```

Comment:
exception does not corresponds to actual error. You array is empty not negative size